portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,25 +36,39 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private CommentLog chatLog = new CommentLog();
-
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("application/json");
-    String json = new Gson().toJson(chatLog);
-    response.getWriter().println(json);
-  }
-
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String currentComment = request.getParameter("comment");
-    chatLog.logComment(currentComment);
+    String comment = request.getParameter("comment");
+    Timestamp ts = new Timestamp(System.currentTimeMillis()); 
+    Date date = new Date(ts.getTime());
+    Entity taskEntity = new Entity("comment");
+    taskEntity.setProperty("text", comment);
+    taskEntity.setProperty("timestamp", date);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(taskEntity);
     response.sendRedirect("/index.html");
   }
 
-  private String convertToJson(List words) {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Object> commentList = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      List<Object> commentItem = new ArrayList<>();
```

Comment:
Rather than having this as list where you have to keep track of what each index is it would be preferable to have this as an object with named properties that you access.